nautilus_trader/adapters/hyperliquid/providers.py

```diff
@@ -15,84 +15,268 @@
 
 from __future__ import annotations
 
-from typing import TYPE_CHECKING, Any
+from collections.abc import Iterable
+from decimal import Decimal
+from typing import Any
 
+from nautilus_trader._libnautilus.hyperliquid import HyperliquidHttpClient
+from nautilus_trader._libnautilus.hyperliquid import HyperliquidInstrumentDef
+from nautilus_trader.adapters.hyperliquid.constants import HYPERLIQUID_VENUE
 from nautilus_trader.common.providers import InstrumentProvider
 from nautilus_trader.config import InstrumentProviderConfig
 from nautilus_trader.core.correctness import PyCondition
 from nautilus_trader.model.identifiers import InstrumentId
-
-
-if TYPE_CHECKING:
-    pass  # TODO: Add imports for actual HyperliquidHttpClient when available
+from nautilus_trader.model.identifiers import Symbol
+from nautilus_trader.model.instruments import CryptoPerpetual
+from nautilus_trader.model.instruments import CurrencyPair
+from nautilus_trader.model.instruments import Instrument
+from nautilus_trader.model.objects import Currency
+from nautilus_trader.model.objects import Price
+from nautilus_trader.model.objects import Quantity
 
 
 class HyperliquidInstrumentProvider(InstrumentProvider):
     """
-    Provides Nautilus instrument definitions from Hyperliquid.
-
-    Parameters
-    ----------
-    client : Any
-        The Hyperliquid HTTP client (to be implemented).
-    config : InstrumentProviderConfig, optional
-        The instrument provider configuration, by default None.
-
+    Load spot and perpetual instruments from Hyperliquid's REST ``/info`` API.
     """
 
     def __init__(
         self,
-        client: Any | None = None,  # TODO: Replace with actual HyperliquidHttpClient when available
+        client: HyperliquidHttpClient,
         config: InstrumentProviderConfig | None = None,
+        *,
+        include_spot: bool = True,
```

Comment:
We typically have some kind of `HyperliquidProductType` or instrument type enum list for this rather than separate bools.